### DISCUSSION TITLE
[Feature request] .h-text and .w-text to match SVG size to font size

### DISCUSSION THREAD

**Discussion by mainej** [tailwindlabs/tailwindcss#3108] [Ideas]
I often use SVG arrows and chevrons inline with other copy. I prefer when these SVGs scale with the surrounding font, because I feel prone to forgetting to change one when changing the other.

This is trivial to do by extending the space utilities with `text: '1em'`. That lets me write `<p class="text-lg"><svg class="h-text w-text" ...` which usually makes SVGs that are approximately the right size, even as I change the font size of the paragraph. If I need more control over a particular SVG, I switch back to explicit sizes like `h-5`. (I don't like using em-based sizing for anything besides `1em`, for the reasons described in #1953.)

Is this technique useful enough and/or common enough to include `text: '1em'` in Tailwind's default spacing config?

**Comment by simonswiss** [comment]
Hey,

That's definitely an interesting idea! 💡

Since a few core plugins use the `spacing` scale, there is the risk of this adding a few extra, unnecessary utilities. This is definitely a good use case, but not sure if this should be implemented in core at this point.

Good news is it's incredibly quick to extend the spacing scale with `"text": "1em"` like you mentioned.

Thanks for sharing - again that's a pretty neat idea to have scalable "inline-with-text" icons 👍

**Reply by mainej** [reply]
Sounds good. Thanks for the feedback! 👍
